torch/distributed/nn/jit/instantiator.py

```diff
@@ -0,0 +1,151 @@
+
+    arg_type_str_list = arg_type_str_list[1:]  # Remove "self".
+    arg_types_str = ", ".join(arg_type_str_list)
+
+    assert len(method_schema.returns) == 1
+    argument = method_schema.returns[0]
+    return_type_str = str(argument.type)
+
+    return args_str, arg_types_str, return_type_str
+
+
+def _write(out_path, text):
+    try:
+        with open(out_path, "r") as f:
+            old_text = f.read()
+    except IOError:
+        old_text = None
+    if old_text != text:
+        with open(out_path, "w") as f:
+            print("Writing {}".format(out_path))
+            f.write(text)
+    else:
+        print("Skipped writing {}".format(out_path))
+
+
+def cleanup_generated_modules():
+    generated_module_paths = pathlib.Path(INSTANTIATED_TEMPLATE_DIR_PATH).glob("_remote_module*.py")
+    for file_path in generated_module_paths:
+        assert file_path.is_file(), f"Epect {file_path} to be a file"
+        print(f"Removing {file_path}")
+        file_path.unlink()
+
+
+def instantiate_remote_module_template(module_interface_cls, is_scriptable):
+    # Generate the template instance name.
+    module_interface_cls_name = torch.jit._qualified_name(module_interface_cls).replace(
+        ".", "_"
+    )
+    generated_module_name = f"_remote_module_{module_interface_cls_name}"
+
+    if is_scriptable:
+        args_str, arg_types_str, return_type_str = get_arg_return_types_from_interface(
+            module_interface_cls
+        )
+        kwargs_str = ""
+        arrow_and_return_type_str = f" -> {return_type_str}"
+        arrow_and_future_return_type_str = f" -> Future[{return_type_str}]"
+    else:
+        args_str = "*args"
+        kwargs_str = "**kwargs"
+        arg_types_str = "*args, **kwargs"
+        arrow_and_return_type_str = ""
+        arrow_and_future_return_type_str = ""
+
+    jit_decorator_str_map = dict(
+        jit_script_decorator="@torch.jit.script",
+        jit_export_decorator="@torch.jit.export",
+        jit_interface_decorator="@torch.jit.interface",
+    )
+    if is_scriptable is False:
+        jit_decorator_str_map = {
+            key: "" for key, value in jit_decorator_str_map.items()
+        }
+
+    remote_forward_template = CodeTemplate.from_file(
+        os.path.join(TEMPLATES_DIR_PATH, "remote_module.py.template")
+    )
+    env = dict(
+        generated_module_name=generated_module_name,
+        module_interface_cls_module_name=module_interface_cls.__module__,
+        module_interface_cls_name=module_interface_cls.__name__,
+        arg_types=arg_types_str,
+        arrow_and_return_type=arrow_and_return_type_str,
+        arrow_and_future_return_type=arrow_and_future_return_type_str,
+        args=args_str,
+        kwargs=kwargs_str,
+        **jit_decorator_str_map,
+    )
+    generated_code_text = remote_forward_template.substitute(env)
+    out_path = os.path.join(
+        INSTANTIATED_TEMPLATE_DIR_PATH, f"{generated_module_name}.py"
+    )
```

Comment:
Can we just use python tempfile (https://docs.python.org/3/library/tempfile.html) for this and rely on python to delete these files?

Reply:
Then, we'll need to create a `__init__.py` there and add it to `sys.path`, does that sound good to you?

Reply:
This won't work. 

Since the TorchScript function `_remote_forward` is referenced by qualifed_name, they must be put at the same python module path on both client and server sides.
However, `tempfile` always generates a random file name.